### PR TITLE
Enhanced IPV6 handling in astats

### PR DESCRIPTION
**Pull request by serDrem**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/22/

File: traffic_server/plugins/astats_over_http/astats_over_http.c [hunk 1]
```diff
@@ -570,22 +574,28 @@ static bool is_ip_allowed(const config_t* config, const struct sockaddr* addr) {
...
 	}
-	const char *ip = addr->sa_data+2;
 
 	if (addr->sa_family == AF_INET && config->allowIps) {
+		const struct sockaddr_in* addr_in = (struct sockaddr_in*) addr;
+		const char *ip = (char*) &addr_in->sin_addr;
+
 		for(i=0; i < config->ipCount; i++) {
-			ipmask = config->allowIps+ (i*5);
+			ipmask = config->allowIps + (i*5);
```
Comment (jrushford) [inline traffic_server/plugins/astats_over_http/astats_over_http.c]: @serDrem you should fix the ipv4 stuff too.  Besides just being bad, (I * 5) is meaningless to the reader.  Should always use sizeof() as thats the correct way to code this plus "i * sizeof(struct in_addr) + 1" lets the reader understand whats being calculated.

Comment (serDrem) [reply traffic_server/plugins/astats_over_http/astats_over_http.c]: done

**Comment by anna-k-1** [comment]
@serDrem Can you please fix your conflicts so we can review this PR? Thanks!

**Comment by ezelkow1** [comment]
Functionality already added to astats with commit
https://github.com/apache/trafficcontrol/commit/1283a52e04d1019c095ea0bda2036caaff9392e7#diff-42ee9311e27e95151959e3950de3b134

**Pull request closed without merging**